### PR TITLE
Proper implementation of usesType method

### PR DESCRIPTION
**Pull request by kornilova-l**
It used to compare pointers, I added `operator==` method to subclasses of `Type`

### CODE REVIEW

File: bindgen/ir/types/ArrayType.h [hunk 1]
```diff
@@ -2,17 +2,20 @@
 #define SCALA_NATIVE_BINDGEN_ARRAYTYPE_H
 
 #include "Type.h"
+#include <stdint-gcc.h>
```
Comment (jonas) [inline bindgen/ir/types/ArrayType.h]: This `#include` breaks compilation on macOS. Doesn't look like it is needed. Is it an IntelliJ auto-import?

Comment (kornilova-l) [reply bindgen/ir/types/ArrayType.h]: Yes, it can be removed

**Comment by jonas** [comment]
Done in #112

File: bindgen/ir/types/ArrayType.h (hunk 1 above)
Comment (jonas) [reply bindgen/ir/types/ArrayType.h]: Done in #112

**Pull request merged**